torch/csrc/jit/runtime/interpreter/code_impl.h

```diff
@@ -213,13 +224,64 @@ struct CodeImpl {
     return instructions_source_;
   }
 
+  NodeSourceInfo getSourceInfoFromSourceRange(const SourceRange& range) {
+    NodeSourceInfo nodeSource;
+
+    SourceRange r = range;
+    if (range.source()) {
+      if (auto orig = range.source()->findSourceRangeThatGenerated(r)) {
```

Comment:
I am wondering about the cost of `findSourceRangeThatGenerated` when called on all lines of code. Can this be run on a model that includes  original source ranges and measured the total loading time with/without this preprocessing step? The code is going to read and unpickle some debug files that would not otherwise have been done, which I think can significantly increase the loading time if those files are big. I didn’t notice before that this pathway is only done lazily but includes file reading.

It would be fine to land this PR without this line, and just record the stack traces without original source code. It would also be fine to measure the overall cost for some big model with source ranges and figure out that only adds a small amount of time to decoding.

Reply:
PR updated to split `findSourceRangeThatGenerated` into a separate PR